restore-hooks_product-requirements.md

```diff
@@ -0,0 +1,309 @@
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 4</span>**
+
+
+    **Title:** Specify container in which to run pre-hook
+
+
+    **Description: **As a user, if I do not want to run the pre-hook command on the first container in the pod (default container), I would like the option to annotate the specific container that i would like the hook to run in.
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 5</span>**
+
+
+    **Title: **Check for latest snapshot 
+
+
+    **Description: **As a user, I would like Velero to run a check for the latest snapshot in object storage prior to starting restore operations on a pod. 
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 6</span>**
+
+
+    **Title: **Display/surface output from restore hooks/restore status 
+
+
+    **Description:** As a user, I would like to see the output of the restore hooks/status of my restore surfaced from the pod volume restore status. Including statuses:  Pending, Running/In Progress, Succeeded, Failed, Unknown.
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 7</span>**
+
+
+    **Title: **Restore metadata
+
+
+    **Description: **As a user, I would like to have the metadata of the contents of what was restored using Velero.
+
+
+    _Note: Kubernetes patterns may result in some snapshot metadata being overwritten during restore operations._
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 8</span>*
+
+    **Title: **Increase default restore and retry limits. 
+
+
+    **Description: **As a user, I would like to increase the default restore retry and timeout limits from the default values to some other value I would like to specify. 
+
+
+    _Note: See use case 11 for the default value specifications. _
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+User Experience
+
+The following is representative of what the user experience could look like for Velero restore pre-hooks and post-hooks. 
+
+**_Note: These examples are representative and are not to be considered for use in pre- and post- restore hook operations until the technical design is complete._**
+
+**Restore Pre-Hooks**
+
+<span style="text-decoration:underline;">Container Command</span>
+
+
+```
+pre.hook.restore.velero.io/container
+
+kubectl patch backupstoragelocation <STORAGE LOCATION NAME> \
```

Comment:
What's the purpose of setting the backup storage location to read only here?

Reply:
I have the same question.